lib/helper/firefoxProfileCreator.py

```diff
@@ -97,8 +97,13 @@ def _install_profile_extensions(self, extensions_settings={}):
                         data['addons'].append(new_data)
 
                         pos = len(data['addons']) - 1
-                        new_xpi_name = data['addons'][pos]['descriptor'].split("/")[-1].split("\\")[-1]
-                        data['addons'][pos]['descriptor'] = os.path.join(extensions_folder, new_xpi_name)
+                        # old add-ons using 'descriptor' to store location while new one is using 'path'
+                        if 'descriptor' in data['addons'][pos]:
+                            new_xpi_name = data['addons'][pos]['descriptor'].split("/")[-1].split("\\")[-1]
+                            data['addons'][pos]['descriptor'] = os.path.join(extensions_folder, new_xpi_name)
+                        elif 'path' in data['addons'][pos]:
+                            new_xpi_name = data['addons'][pos]['path'].split("/")[-1].split("\\")[-1]
+                            data['addons'][pos]['path'] = os.path.join(extensions_folder, new_xpi_name)
```

Comment:
how about using for loop instead of two if ?
for key_name in ['descriptor', 'path']:
    if key_name in data['addons'][pos]:
        new_xpi_name = data['addons'][pos][key_name].split("/")[-1].split("\\")[-1]
        data['addons'][pos][key_name] = os.path.join(extensions_folder, new_xpi_name)
        break